kuma/wiki/forms.py

```diff
@@ -295,6 +305,34 @@ def clean_slug(self):
 
         return slug
 
+    def clean_tags(self):
+        """
+        Validate the tags ensuring we have no case-sensitive duplicates.
+        """
+        tags = self.cleaned_data['tags']
+        dupe_tags = []
+
+        if tags:
+            for tag in parse_tags(tags):
+                # Note: The exact match query doesn't work correctly with
+                # MySQL with regards to case-sensitivity. If we move to
+                # Postgresql in the future this code may need to change.
+                doc_tag = DocumentTag.objects.filter(
+                    name__exact=tag).values_list('name', flat=True)
+                # Write a log we can grep to help find duplicate document tags.
+                if len(doc_tag) > 1:
+                    log.warn('Found duplicate document tags: %s' % doc_tag)
+
+                if doc_tag:
+                    if doc_tag[0] != tag and doc_tag[0].lower() == tag.lower():
+                        dupe_tags.append(doc_tag[0])
+
+            if dupe_tags:
+                raise forms.ValidationError(
+                    TAG_DUPE % {'tag': ', '.join(dupe_tags)})
```

Comment:
For the join of a list that is used in translated strings it would be good to use `django.utils.text.get_text_list` since commas aren't universal signs of combination